src/modules/python/pbs/v1/_pmi_types.py

```diff
@@ -120,13 +120,19 @@ def _check_pmi(self):
     def connect(self, endpoint=None, port=None, job=None):
         self._check_pmi()
         if job is None:
-            job = pbs.event().job
+            try:
+                job = pbs.event().job
+            except:
+                pass
```

Comment:
blanket try/excepts are bad in hooks.  Use a more narrowly focused except.  I believe you are worried about a NameError exception.  An example of why this is bad is that the hook infrastructure will send a KeyboardInterrupt on a hook alarm.  If that happens during your try/except, you'll ignore the alarm.

Reply:
Thanks @bhroam for the valuable suggestion, i have updated the except block now it will handle only "EventIncompatibleError"